### PR TITLE
[ONNX] Update repeat_interleave symbolic

### PR DESCRIPTION
**Pull request by shubhambhokare1**
Add implementation for cases when
- interleaving happens along dim which consist of dynamic axes

### CODE REVIEW

File: torch/onnx/symbolic_opset11.py [hunk 1]
```diff
@@ -830,3 +830,100 @@ def prim_ConstantChunk(g, self, chunks, dim):
...
+    from torch.onnx.symbolic_opset9 import reshape
+    input = self
+    final_dim = dim
+    # if dim is None flatten
+    # By default, use the flattened input array, and return a flat output array
+    if sym_help._is_none(dim):
+        input = reshape(g, self, g.op("Constant", value_t=torch.tensor([-1])))
+        dim = 0
+    else:
+        dim = sym_help._maybe_get_scalar(dim)
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset11.py:844]: Can dim be Value or Negative?

File: torch/onnx/symbolic_opset11.py [hunk 2]
```diff
@@ -830,3 +830,100 @@ def prim_ConstantChunk(g, self, chunks, dim):
...
+    perm_i[0], perm_i[dim] = perm_i[dim], perm_i[0]
+
+    # Cases when repeats is a single value tensor and dim has unknown input size
+    if (repeats_dim == 0 or (repeats_dim == 1 and repeats_sizes[0] == 1)) and input_sizes[dim] == 0:
+        if not sym_help._is_tensor(repeats):
+            repeats = g.op("Constant", value_t=torch.LongTensor(repeats))
+        reps = sym_help._size_helper(g, input, dim)
+        reps = unsqueeze(g, reps, 0)
+        repeats = g.op("Expand", repeats, reps)
+    elif repeats_dim == 1 and input_sizes[dim] == 0:
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset11.py:874]: is `repeats_dim == 1` redundant?

File: torch/onnx/symbolic_opset11.py [hunk 3]
```diff
@@ -830,3 +830,100 @@ def prim_ConstantChunk(g, self, chunks, dim):
...
+    # Loop is of the following pattern
+    # input (trip_count, cond)
+    #   int trip_count = ...;
+    #   bool cond = ...;
+    #   for (int i=0; i < trip_count && cond; ++i) {
+    #     cond = ...;
+    #   }
+
+    # Loop conditions
+    loop_condition = g.op("Constant", value_t=torch.tensor(1))
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset11.py:898]: Wondering if it's possible to create bool constant directly?

File: torch/onnx/symbolic_opset11.py [hunk 4]
```diff
@@ -830,3 +830,100 @@ def prim_ConstantChunk(g, self, chunks, dim):
...
+
+    r_split = loop_block.op("SequenceAt", r_splits, block_input_iter)
+    i_split = loop_block.op("SequenceAt", i_splits, block_input_iter)
+
+    i_split = unsqueeze(loop_block, i_split, dim + 1)
+    r_concat = [loop_block.op("Constant", value_t=torch.LongTensor(input_sizes_temp[:dim + 1])),
+                r_split,
+                loop_block.op("Constant", value_t=torch.LongTensor(input_sizes_temp[dim + 1:]))]
+    r_concat = loop_block.op("Concat", *r_concat, axis_i=0)
+    i_split = expand(loop_block, i_split, r_concat, None)
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset11.py:916]: can `expand` take `r_concat` that contain -1? what about multiple -1?

File: torch/onnx/symbolic_opset11.py [hunk 5]
```diff
@@ -830,3 +830,100 @@ def prim_ConstantChunk(g, self, chunks, dim):
...
+    cond_out = loop_block.op("Cast", loop_condition, to_i=9)
+    _add_output_to_block(loop_block, cond_out)
+    _add_output_to_block(loop_block, i_split)
+    loop_out = loop.node().output()
+
+    # In this loop, the outputs are scan outputs and are concatenated along
+    # the zero'th dimension (by default). In order to avoid this and concatenate
+    # along the dimension provided, some post-processing is required
+    loop_out = g.op("Transpose", loop_out, perm_i=perm_i)
+    return reshape(g, loop_out, g.op("Constant", value_t=torch.LongTensor(input_sizes)))
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset11.py:929]: nit: improve readability by renaming input_sizes and input_sizes_temp. e.g. for this line the input_sizes actually resembles output_sizes

**Comment by BowenBao** [comment]
Thanks, looks good overall, have some questions regarding details

File: torch/onnx/symbolic_opset11.py (hunk 2 above)
Comment (shubhambhokare1) [reply torch/onnx/symbolic_opset11.py:874]: Both are different cases as 
1) [0] ,single value 1-dim tensor
2) [0, 1, 2] ,multi value 1-dim tensor

File: torch/onnx/symbolic_opset11.py (hunk 4 above)
Comment (shubhambhokare1) [reply torch/onnx/symbolic_opset11.py:916]: Yes, it should be able to handle this, as values in input_sizes_temp are being set to -1

**Review by BowenBao — APPROVED** [review]
